fvm/evm/evm_test.go

```diff
@@ -1462,6 +1463,207 @@ func TestDryRun(t *testing.T) {
 			})
 	})
 
+	t.Run("test dry run storing current value", func(t *testing.T) {
+		RunWithNewEnvironment(t,
+			chain, func(
+				ctx fvm.Context,
+				vm fvm.VM,
+				snapshot snapshot.SnapshotTree,
+				testContract *TestContract,
+				testAccount *EOATestAccount,
+			) {
+				data := testContract.MakeCallData(t, "store", big.NewInt(0))
+
+				limit := uint64(math.MaxUint64 - 1)
+				tx := gethTypes.NewTransaction(
+					0,
+					testContract.DeployedAt.ToCommon(),
+					big.NewInt(0),
+					limit,
+					big.NewInt(0),
+					data,
+				)
+				result := dryRunTx(t, tx, ctx, vm, snapshot, testContract)
+				require.Equal(t, types.ErrCodeNoError, result.ErrorCode)
+				require.Equal(t, types.StatusSuccessful, result.Status)
+				require.Greater(t, result.GasConsumed, uint64(0))
+				require.Equal(
+					t,
+					uint64(23873)+params.SstoreSentryGasEIP2200,
```

Comment:
Can we somehow change so we don't use this special value as a hardcoded value, generally not a good idea, if we can somehow calculate it, it will be better

Reply:
That's a good point :100: 
I have re-written the tests in https://github.com/onflow/flow-go/pull/6059/commits/b6fc9e1ea349062b7fbbb9d575421f9d213518e2.
Now, we perform an `EVM.dryRun`, get back the gas estimation, and feed it as the gas limit to `EVM.run`.
We observe that the gas estimation from `EVM.dryRun` is actually higher from the actual gas consumed from `EVM.run`, according to the logic for `SSTORE` opcode gas refunds.